lib/active_merchant/billing/gateways/trans_first.rb

```diff
@@ -40,10 +40,10 @@ def purchase(money, payment, options = {})
         add_amount(post, money)
         add_payment(post, payment)
         add_address(post, options)
-        add_invoice(post, options) if payment.is_a?(CreditCard)
+        add_invoice(post, options) if payment.credit_card?
         add_pair(post, :RefID, options[:order_id], required: true)
 
-        commit((payment.is_a?(Check) ? :purchase_echeck : :purchase), post)
```

Comment:
Yes - I am OK with it for now. At some point however, I would like the method of detecting Check vs. Credit Card to follow the same logic.